client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -264,6 +331,43 @@ const useAddOns = ( siteId?: number, isInSignup = false ): ( AddOnMeta | null )[
 					description,
 				};
 			} );
+	},
+	(
+		state,
+		activeAddOns,
+		spaceUpgradesPurchased,
+		siteId,
+		isLoadingBillingTransactions,
+		mediaStorage
+	) => [
+		getFeaturesBySiteId( state, siteId ),
+		getSiteOption( state, siteId, 'is_commercial' ),
+		state.productsList,
+		activeAddOns,
+		spaceUpgradesPurchased,
+		siteId,
+		isLoadingBillingTransactions,
+		mediaStorage,
+	]
+);
+
+const useAddOns = ( siteId?: number, isInSignup = false ): ( AddOnMeta | null )[] => {
+	// if upgrade is bought - show as manage
+	// if upgrade is not bought - only show it if available storage and if it's larger than previously bought upgrade
+	const { data: mediaStorage } = useMediaStorageQuery( siteId );
+	const { isLoading } = usePastBillingTransactions( isInSignup );
+	const spaceUpgradesPurchased = useSpaceUpgradesPurchased( { isInSignup, siteId } );
+	const activeAddOns = useActiveAddOnsDefs();
+
+	return useSelector( ( state ): ( AddOnMeta | null )[] => {
+		return getAddOnsTransformed(
+			state,
+			activeAddOns,
+			spaceUpgradesPurchased,
+			siteId,
+			isLoading,
+			mediaStorage
+		);
```

Comment:
I confirmed manually this is actually memoized, by simply doing, at various places:
```
let foo;
const useAddOns = ... {
    const addOns = useSelector ...
    console.log( foo === addOns );
    foo = addOns;
    ...
```

I won't create a test for this, being this is something we will refactor next. It would be overkill to mock all the dependencies right now, especially as I doubt there will be any heavy coding in here before the refactor.

However, it would be nice to guard the final shape with a test of the form "should be memorized", to hopefully avoid breaking easily.

@jeyip @southp

Reply:
> However, it would be nice to guard the final shape with a test of the form "should be memorized", to hopefully avoid breaking easily.

Good thinking :) I don't think we need to guard every hook, but we can guard the critical ones once identified. e.g. my gut feeling is that `useAddOns` and `useGridPlans` would be good candidates, but we might need to define "what's critical" first.